src/main/scala/Situation.scala

```diff
@@ -535,8 +564,90 @@ object Situation {
     def toFairySF      = sys.error("Can't make fairysf situation from togyzkumalak situation")
     def toChess        = sys.error("Can't make chess situation from togyzkumalak situation")
     def toDraughts     = sys.error("Can't make draughts situation from togyzkumalak situation")
-    def toSamurai      = sys.error("Can't make draughts situation from samurai situation")
+    def toSamurai      = sys.error("Can't make samurai situation from togyzkumalak situation")
     def toTogyzkumalak = s
+    def toGo           = sys.error("Can't make go situation from togyzkumalak situation")
+  }
+
+  final case class Go(s: go.Situation)
+      extends Situation(
+        Board.Go(s.board),
+        s.player
+      ) {
+
+    lazy val moves: Map[Pos, List[Move]] = Map.empty[Pos, List[Move]]
+
+    lazy val check: Boolean = false
+
+    def checkSquare = None
+
+    def opponentHasInsufficientMaterial: Boolean = s.opponentHasInsufficientMaterial
+
+    def threefoldRepetition: Boolean = false
+
+    def isRepetition: Boolean                    = false
+    override lazy val perpetualPossible: Boolean = false // not allowed to repeat ko
+
+    def end: Boolean = s.end
+
+    def winner: Option[Player] = s.winner
+
+    lazy val destinations: Map[Pos, List[Pos]] = Map.empty[Pos, List[Pos]]
+
+    def drops: Option[List[Pos]] = s.drops.map(_.map(Pos.Go))
+
+    def dropsByRole: Option[Map[Role, List[Pos]]] = s.dropsByRole.map(_.map {
+      case (r: go.Role, p: List[go.Pos]) => (Role.GoRole(r), p.map(Pos.Go))
+    })
+
+    def dropsAsDrops: List[Drop] = s.dropsAsDrops.map(Drop.Go)
+
+    def passes: List[Pass] = pass.fold[List[Pass]](_ => List.empty, p => List(p))
```

Comment:
I dont understand this line. Can you explain it?

Reply:
added a todo to convert other sys.error to validated error messages